packages/cli/src/lib/defaults/deploy-modules/http/index.ts

```diff
@@ -34,28 +32,43 @@ const dirToFormData = (baseDirPath: string) => {
 };
 
 class HTTPDeployer implements DeployModule {
-  async execute(uri: Uri, config?: { postUrl: string }): Promise<Uri> {
-    if (!isValidUri(uri)) {
-      throw new Error(
-        `HTTP Deployer error: Invalid URI: ${uri.toString()}. Supplied URI needs to be a Filesystem URI, example: fs/./build`
-      );
-    }
-
+  async execute(
+    uri: Uri,
+    config?: { postUrl: string; headers: { name: string; value: string }[] }
+  ): Promise<Uri> {
     if (!config?.postUrl) {
       throw new Error(
         `HTTP Deployer error: No postUrl provided. Please provide a postUrl in the deploy manifest's config object`
       );
     }
 
-    const formData = dirToFormData(uri.path);
+    let response;
+    if (uri.authority === "fs") {
```

Comment:
"file" is also being treated as a valid file-system URI authority, going to add that here.